dotnet/sdk-phase-2.md

```diff
@@ -0,0 +1,484 @@
+        Func<Task<TResult>> activity, ActivityStartOptions options) { /*...*/ }
+
+    public static Task<TResult> ExecuteActivityAsync<T, TResult>(
+        Func<T, TResult> activity, T arg, ActivityStartOptions options) { /*...*/ }
+
+    public static Task<TResult> ExecuteActivityAsync<T, TResult>(
+        Func<T, Task<TResult>> activity, T arg, ActivityStartOptions options) { /*...*/ }
+
+    public static Task ExecuteActivityAsync(
+        string activity, object[] args, ActivityStartOptions options) { /*...*/ }
+
+    public static Task<TResult> ExecuteActivityAsync<TResult>(
+        string activity, object[] args, ActivityStartOptions options) { /*...*/ }
+
+    // Child workflows
+
+    public static Task<ChildWorkflowHandle> StartChildWorkflowAsync(
+        Func<Task> workflow, ChildWorkflowStartOptions options) { /*...*/ }
+
+    public static Task<ChildWorkflowHandle> StartChildWorkflowAsync<T>(
+        Func<T, Task> workflow, T arg, ChildWorkflowStartOptions options) { /*...*/ }
+
+    public static Task<ChildWorkflowHandle<TResult>> StartChildWorkflowAsync<TResult>(
+        Func<Task<TResult>> workflow, ChildWorkflowStartOptions options) { /*...*/ }
+
+    public static Task<ChildWorkflowHandle<TResult>> StartChildWorkflowAsync<T, TResult>(
+        Func<T, Task<TResult>> workflow, T arg, ChildWorkflowStartOptions options) { /*...*/ }
+
+    public static Task<ChildWorkflowHandle> StartChildWorkflowAsync(
+        string workflow, object[] args, ChildWorkflowStartOptions options) { /*...*/ }
+
+    public static Task<ChildWorkflowHandle<TResult>> StartChildWorkflowAsync<TResult>(
+        string workflow, object[] args, ChildWorkflowStartOptions options) { /*...*/ }
+
+    public static Task ExecuteChildWorkflowAsync(
+        Func<Task> workflow, ChildWorkflowStartOptions options) { /*...*/ }
+
+    public static Task ExecuteChildWorkflowAsync<T>(
+        Func<T, Task> workflow, T arg, ChildWorkflowStartOptions options) { /*...*/ }
+
+    public static Task<TResult> ExecuteChildWorkflowAsync<TResult>(
+        Func<Task<TResult>> workflow, ChildWorkflowStartOptions options) { /*...*/ }
+
+    public static Task<TResult> ExecuteChildWorkflowAsync<T, TResult>(
+        Func<T, Task<TResult>> workflow, T arg, ChildWorkflowStartOptions options) { /*...*/ }
+
+    public static Task ExecuteChildWorkflowAsync(
+        string workflow, object[] args, ChildWorkflowStartOptions options) { /*...*/ }
+
+    public static Task<TResult> ExecuteChildWorkflowAsync<TResult>(
+        string workflow, object[] args, ChildWorkflowStartOptions options) { /*...*/ }
+
+    // Signal and query handlers
+
+    public bool TryGetSignalHandler(string name, out Delegate handler) { /*...*/ }
+
+    public void SetSignalHandler(string name, Func<Task> handler) { /*...*/ }
+
+    public void SetSignalHandler<T>(string name, Func<T, Task> handler) { /*...*/ }
+
+    public void RemoveSignalHandler(string name) { /*...*/ }
```

Comment:
What happens with this and tryget (as well as for queries) if the handler was defined by `[WorkflowSignal]` ?

Reply:
The handler map is built from the attributes but technically is mutable through these methods. So yes, you can technically remove a handler added via an attribute. And of course getting a handler by name works no matter how that handler was added. In that sense attribute-based handlers (our preference) are just sugar on this. It doesn't make much sense for a user to remove an attribute-added handler IMO, but at least it's a simple to understand concept. This is the same way it works in Python atm too. I hope nobody ever calls dynamic handler things FWIW.